internal/testing/setup/setup.go

```diff
@@ -100,15 +101,19 @@ func awsSession(region string, client *http.Client) (*session.Session, error) {
 // from files. rf is a modifier function that will be invoked with the address
 // of the httpreplay.Recorder object used to obtain the client; this function
 // can mutate the recorder to add provider-specific header filters, for example.
-func NewRecordReplayClient(ctx context.Context, t *testing.T, rf func(r *httpreplay.Recorder), opts ...option.ClientOption) (c *http.Client, cleanup func()) {
+// An initial state is returned for tests that need a state to have
+// deterministic results.
+func NewRecordReplayClient(ctx context.Context, t *testing.T, rf func(r *httpreplay.Recorder), opts ...option.ClientOption) (*http.Client, func(), int64) {
 	httpreplay.DebugHeaders()
 	path := filepath.Join("testdata", t.Name()+".replay")
 	if *Record {
 		t.Logf("Recording into golden file %s", path)
 		if err := os.MkdirAll(filepath.Dir(path), 0755); err != nil {
 			t.Fatal(err)
 		}
-		rec, err := httpreplay.NewRecorder(path, nil)
+		state := time.Now()
+		b, _ := state.MarshalBinary()
+		rec, err := httpreplay.NewRecorder(path, b)
```

Comment:
Move the error check on line 118 above the call to rf.

Reply:
Done.